* transferring Globus to S3 
http://docs.staging.globuscs.info/premium-storage-connectors/legacy/aws-s3-legacy/
http://docs.staging.globuscs.info/premium-storage-connectors/aws-s3/
* short RNA-Seq
** encode
file encode-rnaseq.status.tsv in get is used to track the status
in build/t2tChm13_20200727/shortRnaSeq

*** set 1 2020-08-18
- poly(A) RNA, 
- tissues: brain, testis, liver, H1, GM12878

# in git: t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq
~/compbio/gencode/projs/icedb/gencode-icedb-dev/bin/rslEncodeDccQuery --assay=polyA_RNA --tissue=brain --tissue=testis --tissue=liver  --tissue=H1 --tissue=GM12878 encode-rnaseq.set1.tsv

68 runs in 28 experiments


cd build/t2tChm13_20200727/shortRnaSeq/tmp
 ~/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq/encode-rnaseq.set1.tsv ..

# create genome
mkdir genome
zcat ../../../../../build/t2tChm13_20200727/genome/t2tChm13_20200727.fa.gz >genome/t2tChm13_20200727.fa
nice STAR --runMode genomeGenerate --runThreadN 64 --genomeFastaFiles genome/t2tChm13_20200727.fa --genomeDir genome/star --outFileNamePrefix genome/star  &

# generate commands:
tawk 'NR>1{print "starAlign " $2".raw.bam genome/star " $10 " " $14}'  ../../../../t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq/encode-rnaseq.set1.tsv  >../runSet1Cmds

# for running on parasol needs 32gb
ln -s ../../../../t2t-chm13-gene-analysis/bin/starAlignPara 
para create -ram=32g jobs1 -batch=b1
para push -ram=32g -batch=b1

*** set 2 2020-08-23
# get second set of RNAs, get  ploy-A min 100 bases
~/compbio/gencode/projs/icedb/gencode-icedb-dev/bin/rslEncodeDccQuery --assay=polyA_RNA --minReadLength=100  tmp.tsv

selectById -tsv -not 2 encode-rnaseq.set1.tsv 2 tmp.tsv >encode-rnaseq.set2.tsv
encode-rnaseq.set2.tsv  447
# add this to status list
tawk 'NR>1{print $2, "pending", ""}' encode-rnaseq.set2.tsv  >x

*** all sets of unfiltered mappings in 
/team-genes/20200727/shortRnaSeq/raw-mappings

** intron calling: 
*** Run by Ann Mc Cartney using intronProspector 0.6.o with commands in the form:
  intronProspector --genome-fasta=t2tChm13_20200727.fa.gz  --min-confidence-score=1.0 --intron-calls=$run.raw.introns.tsv  ../raw/$run.raw.bam
on set1 and set2

*** /team-genes/20200727/shortRnaSeq/raw-introns/

** generate combined TSV
cd shortRnaSeq/

export PATH=${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${HOME}/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/bin/:${PATH}

mkdir /data/tmp/markd/introns
intronMergeGenMake --inputsPerJob=5 raw-introns /data/tmp/markd/introns shortRnaSeq-2020-08-28 
cd /data/tmp/markd/introns
nice make -j 32 >&log&

* longRnaSeq:

