* comparing refseq gene blat alignments to V0.7 and V1.0
PATH=/hive/users/markd/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/bin:${PATH}

* HGNC data
wget -nv 'https://www.genenames.org/cgi-bin/genegroup/download-all?format=json'
json_pp -json_opt  pretty,indent,utf8,allow_bignum <download-all?format=json >gene-groups.json

* tighter filtering of alignments
cd build/paper-analysis/2021-01/refSeqBlat
ln -s ../../../../../../build/t2t-chm13-v0.7/refSeqBlat/data
ln -s ../../../../../../build


filt=filt1
mkdir $filt
pslCDnaFilter -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -globalNearBest=0.001 -minId=0.95 -minCover=0.85 -filterWeirdOverlapped -statsOut=${filt}/refSeq-v1.0.${filt}.fstats build/t2t-chm13-v1.0/refSeqBlat/refSeq-curated.raw.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >${filt}/refSeq-v1.0.${filt}.psl.gz&
pslCDnaFilter -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -globalNearBest=0.001 -minId=0.95 -minCover=0.85 -filterWeirdOverlapped -statsOut=${filt}/refSeq-v0.7.${filt}.fstats build/t2t-chm13-v0.7/refSeqBlat/refSeq-curated.raw.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >${filt}/refSeq-v0.7.${filt}.psl.gz&
pslCDnaFilter -repsAsMatch -ignoreNs -bestOverlap -polyASizes=data/GRCh38_latest_rna.polya.gz  -globalNearBest=0.001 -minId=0.95 -minCover=0.85 -filterWeirdOverlapped -statsOut=${filt}/refSeq-GRCh38.${filt}.fstats build/GRCh38/refSeqBlat/refSeq-curated.raw.gz stdout | sort -k 14,14 -k 16,16n |pigz -9c >${filt}/refSeq-GRCh38.${filt}.psl.gz&


* get stats on alignments

# these steps parameterized by filt
filt=filt1

for dset in v1.0 v0.7 GRCh38 ; do
    pslStats -queryStats -queries=data/refSeq-curated.sizes ${filt}/refSeq-${dset}.${filt}.psl.gz ${filt}/refSeq-${dset}.${filt}.qstats &
    pslStats -queries=data/refSeq-curated.sizes ${filt}/refSeq-${dset}.${filt}.psl.gz ${filt}/refSeq-${dset}.${filt}.stats &
    pslStats -overallStats -queries=data/refSeq-curated.sizes ${filt}/refSeq-${dset}.${filt}.psl.gz ${filt}/refSeq-${dset}.${filt}.allstats &
    mrnaToGene -quiet -genePredExt -cdsFile=data/refSeq-curated.cds.gz -insertMergeSize=0 ${filt}/refSeq-${dset}.${filt}.psl.gz stdout | pigz -c >${filt}/refSeq-${dset}.${filt}.gp.gz &
    overlapSelect -statsOutput -strand -excludeSelf ${filt}/refSeq-${dset}.${filt}.psl.gz ${filt}/refSeq-${dset}.${filt}.psl.gz stdout | pigz -c > ${filt}/refSeq-${dset}.${filt}.overstats.gz &
done

* frameshift data
for dset in v1.0 v0.7 GRCh38 ; do
    blatFrameShiftInfo ${filt}/refSeq-${dset}.${filt}.gp.gz ${filt}/refSeq-${dset}.${filt}.frameshifts.bed &
done

* reports
for dset in v1.0 v0.7 GRCh38 ; do
    blatMappingReport --dset=${dset} data/refSeq-curated.meta.tsv.gz ${filt}/refSeq-${dset}.${filt}.stats ${filt}/refSeq-${dset}.${filt}.overstats.gz ${filt}/refSeq-${dset}.${filt}.frameshifts.bed ${filt}/refSeq-${dset}.${filt}.report.tsv &
done
mlr --tsv join -j type,category -f ${filt}/refSeq-v0.7.${filt}.report.tsv  ${filt}/refSeq-v1.0.${filt}.report.tsv \
  | mlr --tsv join -j type,category -f /dev/stdin ${filt}/refSeq-GRCh38.${filt}.report.tsv > ${filt}/refSeq-combined.${filt}.report.tsv


* report explanation;
** type
- trans - all transcripts
- gene - per-gene, based on canonical transcript (MANE select or longest RNA)
** category
- total - total number of RNA
- aligned - total number with at least one alignment
- cover98 - numbers with >= 98% coverage (bases aligned) for best alignment
- cover95 - >=95%, < 98% coverage
- cover75 - >=75%, < 95% coverage
- cover50 - >=50%, < 95% coverage
- coverLess - <50% coverage
- multiAligned - number with multiple alignments after filtering
- paralogAligned - number that overlap another gene
- frameShifted - number with at least one frameshift
- paralogFrameShifted - number that are both paralog and frameshifted

** count and rates for V0.7, v1.0, and GRCh38

* for Ann chrom images
iso-seq https://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.0/CAT/annotation.bb
transmap https://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.0/CAT/filtered_transmap.bb
proseq https://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.0/proSeq/CHM13-5A_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam
       https://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.0/proSeq/CHM13-5B_proseq_cutadapt-2_bt2-vs_filtered_STAR-MDiekhansParam-chm13-20200922_Aligned_sort.out.bam
salmon  https://t2t.gi.ucsc.edu/chm13/hub/t2t-chm13-v1.0/chm13.CATv1.salmonQuant.bb

* gene colored bed for graphic
build/paper-analysis/2021-01/geneColorsBed/

** tmp work to figure things out
mlr --tsv cut -f gene_biotype,transcript_biotype,valid_start,valid_stop CAT/consensus_gene_set/CHM13.gp_info | sort -u >tmp/gp_info.types
mlr --tsv filter '$gene_biotype=="StringTie"' CAT/consensus_gene_set/CHM13.gp_info > tmp/StingTie.gp_info
selectById 2 tmp/StingTie.gp_info 1 CAT/consensus_gene_set/CHM13.gp >tmp/StingTie.gp
