* data
** transferring Globus to S3 
http://docs.staging.globuscs.info/premium-storage-connectors/legacy/aws-s3-legacy/
http://docs.staging.globuscs.info/premium-storage-connectors/aws-s3/
** short RNA-Seq
*** encode
file encode-rnaseq.status.tsv in get is used to track the status
in build/t2tChm13_20200727/shortRnaSeq

**** set 1 2020-08-18
- poly(A) RNA, 
- tissues: brain, testis, liver, H1, GM12878

# in git: t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq
~/compbio/gencode/projs/icedb/gencode-icedb-dev/bin/rslEncodeDccQuery --assay=polyA_RNA --tissue=brain --tissue=testis --tissue=liver  --tissue=H1 --tissue=GM12878 encode-rnaseq.set1.tsv

68 runs in 28 experiments


cd build/t2tChm13_20200727/shortRnaSeq/tmp
 ~/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq/encode-rnaseq.set1.tsv ..

# create genome
mkdir genome
zcat ../../../../../build/t2tChm13_20200727/genome/t2tChm13_20200727.fa.gz >genome/t2tChm13_20200727.fa
nice STAR --runMode genomeGenerate --runThreadN 64 --genomeFastaFiles genome/t2tChm13_20200727.fa --genomeDir genome/star --outFileNamePrefix genome/star  &

# generate commands:
tawk 'NR>1{print "starAlign " $2".raw.bam genome/star " $10 " " $14}'  ../../../../t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq/encode-rnaseq.set1.tsv  >../runSet1Cmds

# for running on parasol needs 32gb
ln -s ../../../../t2t-chm13-gene-analysis/bin/starAlignPara 
para create -ram=32g jobs1 -batch=b1
para push -ram=32g -batch=b1

**** set 2 2020-08-23
# get second set of RNAs, get  ploy-A min 100 bases
~/compbio/gencode/projs/icedb/gencode-icedb-dev/bin/rslEncodeDccQuery --assay=polyA_RNA --minReadLength=100  tmp.tsv

selectById -tsv -not 2 encode-rnaseq.set1.tsv 2 tmp.tsv >encode-rnaseq.set2.tsv
encode-rnaseq.set2.tsv  447
# add this to status list
tawk 'NR>1{print $2, "pending", ""}' encode-rnaseq.set2.tsv  >x

**** all sets of unfiltered mappings in 
/team-genes/20200727/shortRnaSeq/raw-mappings

*** intron calling: 
**** Run by Ann Mc Cartney using intronProspector 0.6.o with commands in the form:
  intronProspector --genome-fasta=t2tChm13_20200727.fa.gz  --min-confidence-score=1.0 --intron-calls=$run.raw.introns.tsv  ../raw/$run.raw.bam
on set1 and set2

**** /team-genes/20200727/shortRnaSeq/raw-introns/

*** generate combined TSV
cd shortRnaSeq/

export PATH=${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${HOME}/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/bin/:${PATH}

mkdir /data/tmp/markd/introns
intronMergeGenMake --inputsPerJob=5 raw-introns /data/tmp/markd/introns shortRnaSeq-2020-08-28 
cd /data/tmp/markd/introns
nice make -j 32 >&log&

** longRnaSeq:

** refseqs not in GRCh38

** gene family data  (geneFamilies)
DEAD END with MULTI FAMILY:

*** data
- Ann obtained list of gene families from ensembl
  ensembl_multigenefamilies_GRCh38_FINAL.tab.gz
- Note, it contains likes like:
   ENSG00000260371	ENSG00000272617	gene_split	ENSG00000213380	
- and line
   Gene stable ID	Human paralogue gene stable ID	Human paralogue homology type
  dropping these for now
*** build
cd geneFamilies/data
geneBoundsFromBigBlatPsl ../../../../build/t2tChm13_20200727/refSeqBlat/refSeq-curated.bigBlatPsl refSeq-curated.genes.bed
geneBoundsFromBigBlatPsl ../../../../build/t2tChm13_20200727/gencodeBlat/gencode.bigBlatPsl gencode.genes.bed

geneFamilyBuildInfo ensembl_multigenefamilies_GRCh38_FINAL.tab.gz multigenefamilies.json.gz

geneFamilyExpansion multigenefamilies.json.gz gencode.genes.bed /dev/null >&exp.out
didn't find any expansion, although appears to be working


** CAT proteins for Ann
bigGenePredToGenePred ../../../../build/t2tChm13_20200727/CAT/assemblyHub/CHM13/consensus.bb  stdout |tawk '$6<$7' |genePredToProt stdin ../../../../build/t2tChm13_20200727/genome/t2tChm13_20200727.2bit cat-consensus-proteins.fa
bgzip --threads=64 cat-consensus-proteins.fa 



* analysis

** geneDup
- Look a genes that have multiple alignments blat alignments in regions with

*** data
% cd geneDup
% geneBoundsFromBigBlatPsl --nameField=geneSym  --geneType=lncRNA --geneType=protein_coding ../../../../build/t2tChm13_20200727/gencodeBlat/gencode.bigBlatPsl data/gencode.lnc-prot.genes.bed

# select to avoid small loci and very large number of duplication
geneDupSelect --minDups=0 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.10k.bed 
geneDupSelect --minDups=2 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.dup2_10k.bed 
geneDupSelect --minDups=4 --maxDups=8 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.dup4-8_10k.bed 
geneDupSelect --minDups=2 --maxDups=8 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.dup2-8_10k.bed 

10k          65174
dup2_10k     45841
dup2-8_10k    5634
dup4-8_10k    2423

*** centromeres
cd centormeres/
- any gene in centromeres that are not aligned in primary assembly
% overlapSelect -inCoordCols=0,1,2 ../data/cenRegions.bed ../selects/gencode.10k.bed centoGenes.bed
  centoGenes 4992

% overlapSelect -inCoordCols=0,1,2 ../../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOver.bed.gz centoGenes.bed centoGenes.unaligned.bed
% overlapSelect -inCoordCols=0,1,2 ../../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOverPrimary.bed.gz centoGenes.bed centoGenes.unaligned-primary.bed
  centoGenes.unaligned 3695
  centoGenes.unaligned-primary 3900 


- centromere browser
     ~/public_html/t2t/genes-20200727/centromere/ctracks.bed
      https://hgwdev.gi.ucsc.edu/~markd/t2t/genes-20200727/centromere/ctracks/

  mkdir ~/public_html/t2t/genes-20200727/centromere
  bedToHtmlDir -hub https://t2t.gi.ucsc.edu/chm13/hub/hub.txt -custom-track https://hgwdev.gi.ucsc.edu/~markd/t2t/genes-20200727/centromere/ctracks.bed -title "Centromere regions t2tChm13_20200727" t2t-chm13.20200727 data/cenRegions.bed ~/public_html/t2t/genes-20200727/centromere


Item: AL590807.1
Position: chr1:118361330-119720545
Genomic Size: 1359215
Strand: +



** intersections:
lastz unaligned
cetromeres
short introns
CAT annotations
lastz transmap



* observations
** gencode/refseq blat align different family members  than CAT
chr3:99,053,792-99,056,919
in seg dup

** chr22 centomere
chr22:458,239-6,761,699




* future:
** chr1:129,035,884-129,059,013 [bogus]
- repetitive array of isoseq 
- 1q21.1 - pass on to David and Colleen
- MD:the PB signal is less clear
  augustus is extending theses in some kind of interesting ways
  It is pretty easy for a gene finding to make up an ORF in a GC region
