* data

export PATH=${HOME}/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/bin:${HOME}/compbio/nanopore/projs/t2t-chm13/T2T-CHM13-hub/bin:${PATH}

** short RNA-Seq
*** encode
file encode-rnaseq.status.tsv in get is used to track the status
in build/t2tChm13_20200727/shortRnaSeq

**** set 1 2020-08-18
- poly(A) RNA, 
- tissues: brain, testis, liver, H1, GM12878

# in git: t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq
~/compbio/gencode/projs/icedb/gencode-icedb-dev/bin/rslEncodeDccQuery --assay=polyA_RNA --tissue=brain --tissue=testis --tissue=liver  --tissue=H1 --tissue=GM12878 encode-rnaseq.set1.tsv

68 runs in 28 experiments


cd build/t2tChm13_20200727/shortRnaSeq/tmp
 ~/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq/encode-rnaseq.set1.tsv ..

# create genome
mkdir genome
zcat ../../../../../build/t2tChm13_20200727/genome/t2tChm13_20200727.fa.gz >genome/t2tChm13_20200727.fa
nice STAR --runMode genomeGenerate --runThreadN 64 --genomeFastaFiles genome/t2tChm13_20200727.fa --genomeDir genome/star --outFileNamePrefix genome/star  &

# generate commands:
tawk 'NR>1{print "starAlign " $2".raw.bam genome/star " $10 " " $14}'  ../../../../t2t-chm13-gene-analysis/build/t2tChm13_20200727/shortRnaSeq/encode-rnaseq.set1.tsv  >../runSet1Cmds

# for running on parasol needs 32gb
ln -s ../../../../t2t-chm13-gene-analysis/bin/starAlignPara 
para create -ram=32g jobs1 -batch=b1
para push -ram=32g -batch=b1

**** set 2 2020-08-23
# get second set of RNAs, get  ploy-A min 100 bases
~/compbio/gencode/projs/icedb/gencode-icedb-dev/bin/rslEncodeDccQuery --assay=polyA_RNA --minReadLength=100  tmp.tsv

selectById -tsv -not 2 encode-rnaseq.set1.tsv 2 tmp.tsv >encode-rnaseq.set2.tsv
encode-rnaseq.set2.tsv  447
# add this to status list
tawk 'NR>1{print $2, "pending", ""}' encode-rnaseq.set2.tsv  >x

**** all sets of unfiltered mappings in 
/team-genes/20200727/shortRnaSeq/raw-mappings

*** intron calling: 
**** Run by Ann Mc Cartney using intronProspector 0.6.o with commands in the form:
intronProspector --genome-fasta=t2tChm13_20200727.fa.gz  --min-confidence-score=1.0 --intron-calls=$run.raw.introns.tsv  ../raw/$run.raw.bam
on set1 and set2

**** /team-genes/20200727/shortRnaSeq/raw-introns/

*** generate combined TSV
cd shortRnaSeq/

export PATH=${HOME}/compbio/gencode/projs/icedb/intronProspector/bin:${HOME}/compbio/nanopore/projs/t2t-chm13/gene-annotation/t2t-chm13-gene-analysis/bin/:${PATH}

mkdir /data/tmp/markd/introns
intronMergeGenMake --inputsPerJob=5 raw-introns /data/tmp/markd/introns shortRnaSeq-2020-08-28 
cd /data/tmp/markd/introns
nice make -j 32 >&log&

** longRnaSeq:
team-genes/isoseq/UW_20200727_alignments - Glennis Logsdon 

** refseqs not in GRCh38

** gene family data  (geneFamilies)
DEAD END with MULTI FAMILY:

*** data
- Ann obtained list of gene families from ensembl
  ensembl_multigenefamilies_GRCh38_FINAL.tab.gz
- Note, it contains likes like:
   ENSG00000260371	ENSG00000272617	gene_split	ENSG00000213380	
- and line
   Gene stable ID	Human paralogue gene stable ID	Human paralogue homology type
  dropping these for now
*** build
cd geneFamilies/data
geneBoundsFromBigBlatPsl ../../../../build/t2tChm13_20200727/refSeqBlat/refSeq-curated.bigBlatPsl refSeq-curated.genes.bed
geneBoundsFromBigBlatPsl ../../../../build/t2tChm13_20200727/gencodeBlat/gencode.bigBlatPsl gencode.genes.bed

geneFamilyBuildInfo ensembl_multigenefamilies_GRCh38_FINAL.tab.gz multigenefamilies.json.gz

geneFamilyExpansion multigenefamilies.json.gz gencode.genes.bed /dev/null >&exp.out
didn't find any expansion, although appears to be working


** CAT proteins for Ann
# get multiexon ones due to FPs in CAT because of low-quility exon
bigGenePredToGenePred ../../../../build/t2tChm13_20200727/CAT/assemblyHub/CHM13/consensus.bb  stdout |tawk '$6<$7 && $8>1' |genePredToProt stdin ../../../../build/t2tChm13_20200727/genome/t2tChm13_20200727.2bit cat-consensus-proteins.meg.fa
bgzip --threads=64 cat-consensus-proteins.meg.fa 


* analysis

** geneDup
- Look a genes that have multiple alignments blat alignments in regions with

*** data
% cd geneDup
% genePredToBed /hive/data/genomes/hg38/bed/gencodeV33/data/gencode.gp data/gencode.srcsizes.bed

# only using multi-exon source genes, not using alignments with large expansion
% geneBoundsFromBigBlatPsl --nameField=geneSym  --geneType=lncRNA --geneType=protein_coding --maxSourceExpansion=1.5 --minExons=2 --sourceAnnot=data/gencode.srcsizes.bed  ../../../../build/t2tChm13_20200727/gencodeBlat/gencode.bigBlatPsl data/gencode.lnc-prot.genes.bed

#+BEGIN_SRC 
# select to avoid small loci, very large number of duplication
geneDupSelect --minDups=0 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.10k.bed 
geneDupSelect --minDups=2 --minMeanSize=10000 --minDups=0 data/gencode.lnc-prot.genes.bed selects/gencode.dup2_10k.bed 
geneDupSelect --minDups=4 --maxDups=8 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.dup4-8_10k.bed 
geneDupSelect --minDups=2 --maxDups=8 --minMeanSize=10000 data/gencode.lnc-prot.genes.bed selects/gencode.dup2-8_10k.bed 
#+END_SRC

10k.bed	32822
dup2_10k.bed	13465
dup2-8_10k.bed	4148
dup4-8_10k.bed	1512

** centromeres
- any gene in centromeres that are not aligned in primary assembly
# 100% in centromere regions, 80% overlap others
#+BEGIN_SRC
cd centromeres
overlapSelect -inCoordCols=0,1,2 -overlapThreshold=1.0 ../data/cenRegions.bed ../selects/gencode.10k.bed 10k/centoGenes.bed
overlapSelect -inCoordCols=0,1,2 -overlapThreshold=0.8 ../../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOver.bed.gz 10k/centoGenes.bed 10k/centoGenes.new.bed
overlapSelect -inCoordCols=0,1,2 -overlapThreshold=0.8 ../../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOverPrimary.bed.gz 10k/centoGenes.bed 10k/centoGenes.new-pri.bed
#+END_SRC

10k/centoGenes	2302
10k/centoGenes.new	300
10k/centoGenes.new-pri	484

../bin/mkCentroTracks 

WEIGHT BY NUMBER OF MAPPED TRANSCRIPTS

** centromere browser
    ~/public_html/t2t/genes-20200727/centromere/ctracks.bed
     https://hgwdev.gi.ucsc.edu/~markd/t2t/genes-20200727/centromere/ctracks/

  mkdir ~/public_html/t2t/genes-20200727/centromere
  bedToHtmlDir -hub https://t2t.gi.ucsc.edu/chm13/hub/hub.txt -custom-track https://hgwdev.gi.ucsc.edu/~markd/t2t/genes-20200727/centromere/ctracks.bed -title "Centromere regions t2tChm13_20200727" t2tChm13_20200727 ../data/cenRegions.bed ~/public_html/t2t/genes-20200727/centromere

    Error line 3541 of https://hgwdev.gi.ucsc.edu/~markd/t2t/genes-20200727/centromere/ctracks.bed: 

chromEnd larger than chrom chr20 size (66202773 > 64444167)
chr20	66187167	66202773	WASHC1	6	+	WASHC1	HGNC:24361	ENSG00000181404.17	protein_coding

** genes in centromere regions:

bigBedToBed ../../../../build/t2tChm13_20200727/CAT/assemblyHub/CHM13/consensus.bb consensus.bed

# or together unique regions
featureBits -countGaps -or -bed=combined.uniq.bed -chromSize=../../../../build/t2tChm13_20200727/uniqRegions/lastz/t2tChm13_20200727.sizes3 nodb ../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOver.bed.gz ../../../../build/t2tChm13_20200727/uniqRegions/cactus/cactus-unique.bed 
217689342 bases of 3056841030 (7.121%) in intersection

cut -f 1 ../geneDup/data/gencode.lnc-prot.genes.bed  |sort -u

** global blat compared in unique regions
#+BEGIN_SRC
cd global
overlapSelect -inCoordCols=0,1,2 -overlapThreshold=0.8 ../../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOver.bed.gz ../selects/gencode.10k.bed 10k/globalGenes.new.bed
overlapSelect -inCoordCols=0,1,2 -overlapThreshold=0.8 ../../../../../build/t2tChm13_20200727/uniqRegions/lastz/lastzUnalignedOverPrimary.bed.gz ../selects/gencode.10k.bed 10k/globalGenes.new-pri.bed
#+END_SRC

../bin/mkGlobalTracks

** things to look at:
lastz unaligned
cetromeres
short introns
CAT annotations
lastz transmap




* centromere observations
** gencode/refseq blat align different family members than CAT
chr3:99,053,792-99,056,919
in seg dup

** MORC4 ENSG00000133131.15 / ENST00000604604.1
- 96% ident, but no other alignments
- chr3:89537725-89947865 (score 293??)
- hg38:  chrX:106,813,871-106,993,340
  suspect GENCODE; long isoform, overlapping 
  EST BF203734 NIH_MGC_17
  EST BF203688 NIH_MGC_17
  mRNA BC009313  NIH_MGC_17

** lncRNAs uniquely align to a different locations
chr15:2,317,241-2,319,011

- hg38
  ENST00000619252.4 (CU633967.1) chr21:5553637-5590200

* missingGenes
** data
generate gencode data (via bigPsl, because the tool isn't general enough)
#+BEGIN_SRC
blddir=../../../../build/t2tChm13_20200727/
hgsql hg38 -Ne 'select * from wgEncodeGencodeCompV33;select * from wgEncodeGencodePseudoGeneV33;' | cut -f 2- >data/gencode.hg38.gp
genePredToFakePsl no data/gencode.hg38.gp -chromSize=/hive/data/genomes/hg38/chrom.sizes data/gencode.hg38.psl data/gencode.hg38.cds
genePredToBed data/gencode.hg38.gp data/gencode.hg38.bed
hgsql hg38 -e 'select ga.transcriptId id, ga.geneName as geneSym, ga.geneId, gs.geneId as hgncId, ga.geneType from wgEncodeGencodeAttrsV33 ga left join wgEncodeGencodeGeneSymbolV33 gs on gs.transcriptId = ga.transcriptId' >data/gencode.hg38.meta.tsv
pslToBigPsl -cds=data/gencode.hg38.cds data/gencode.hg38.psl stdout | sort -k1,1 -k2,2n > tmp/g.bigin1
bedToBigBed -type=bed12+17 -tab -as=${blddir=}/../../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/g.bigin2 /hive/data/genomes/hg38/chrom.sizes data/gencode.hg38.bigPsl
bigBlatPslAddColumns data/gencode.hg38.meta.tsv tmp/g.bigin1 tmp/g.bigin2gzip data/gencode.hg38.meg-lnc-prot.genes.bed
/geneBoundsFromBigBlatPsl --nameField=geneSym  --geneType=lncRNA --geneType=protein_coding --maxSourceExpansion=1.5 --minExons=2 --sourceAnnot=data/gencode.hg38.bed data/gencode.hg38.bigPsl data/gencode.hg38.meg-lnc-prot.genes.bed
#+END_SRC

get data for consensus
need to convert bigGenePred to PSL indexed by ENST
    name= $4
    sourceGene = $21
    sourceTranscript = $22, ENST or N/A
    geneId = $27
    geneName = $19
    37 columns

#+BEGIN_SRC 
bigBedToBed ${blddir}/CAT/assemblyHub/CHM13/consensus.bb stdout | tawk '$22!="N/A"{$4=$22}{print}' >tmp/bigGp
bedToBigBed -type=bed12+25 -tab -as=consensus.as tmp/bigGp ${blddir}/genome/t2tChm13_20200727.sizes tmp/cconsensus.bigGp
bigGenePredToGenePred tmp/cconsensus.bigGp tmp/consensus.gp
genePredToFakePsl -chromSize=${blddir}/genome/t2tChm13_20200727.sizes -qSizes=/hive/data/genomes/hg38/chrom.sizes no tmp/consensus.gp tmp/consensus.psl tmp/consensus.cds
pslToBigPsl -cds=tmp/consensus.cds tmp/consensus.psl stdout | sort -k1,1 -k2,2n > tmp/cat.bigPslIn
bigBlatPslAddColumns data/gencode.hg38.meta.tsv tmp/cat.bigPslIn  tmp/cat.bigPslIn2
bedToBigBed -type=bed12+17 -tab -as=${blddir}/../../T2T-CHM13-hub/etc/bigBlatPsl.as tmp/cat.bigPslIn2 ${blddir}/genome/t2tChm13_20200727.sizes data/consensus.chm13.bigPsl
geneBoundsFromBigBlatPsl --nameField=geneSym  --geneType=lncRNA --geneType=protein_coding --maxSourceExpansion=1.5 --minExons=2 --sourceAnnot=data/gencode.hg38.bed data/consensus.chm13.bigPsl data/consensus.chm13.meg-lnc-prot.genes.bed
#+END_SEC


* CAT
** getting missing genes
ln -s ../../../../../build/t2tChm13_20200727/CAT/databases .

import pandas as pd
import tools.sqlInterface
ref_df = tools.sqlInterface.load_annotation(args.ref_db_path)
tm_filter_eval = tools.sqlInterface.load_filter_evaluation(args.db_path)
merged = ref_df.merge(tm_filter_eval, on=["TranscriptId", "GeneId"], how="left")
missing = merged[merged.AlignmentId.isnull()]
missing_lnc_prot = missing[missing.GeneBiotype.isin(["protein_coding", "lncRNA"])]

raw SQL, you are joining the tables TransMapFilterEvaluation in $genome.db with annotation in $refdb

SELECT name FROM hg38.sqlite_master WHERE type='table';

attach database "databases/GRCh38.db" as hg38;
SELECT FROM TransMapFilterEvaluation tmfe, hg38.annotation ann


tmfe.AlignmentId IS NULL




* proteinCodingUnplaced
Ann's cases of coding intersecting unplaced aligned to T2T
cut -f 1-6 proteincodinginunplacedGRCh38.bed >proteincodinginunplacedGRCh38.bed6
bedToHtmlDir -loc-label-name -hub https://t2t.gi.ucsc.edu/chm13/hub/hub.txt -public-browser -title "Protein coding in unplaced (minimap)" t2tChm13_20200727 proteincodinginunplacedGRCh38.bed6  ~/public_html/t2t/for-ann/proteinCodingUnplaced/browser
https://hgwdev.gi.ucsc.edu/~markd/t2t/for-ann/proteinCodingUnplaced/browser/

** cases
** chr1:16158681-16211287	NBPF10	0	-	NBPF10	HGNC:31992	ENSG00000271425.9	protein_coding -> GO TERM: Cytoplasm -> Full name :Neuroblastoma breakpoint family member 10)
** chr11:2035066-2072180	MRPL23	0	+	MRPL23	HGNC:10322	ENSG00000214026.11	protein_coding -> Placed in 2 places in GRCh38 but only 1 in CHM13 -> mitochondrial Ribosomal protein (involved in mitocondrial translation)
known problem in GRCh38 - random was duplicate of the assembly
** chr15:19505915-19586667	AC134980.3	0	+	AC134980.3	NULL	ENSG00000285472.1	protein_coding
chr15_KI270727v1_random:296,519-448,247
got have GRC incidents
** chr15:19535114-19586674	LINC02203	0	+	LINC02203	HGNC:53069	ENSG00000280709.2	protein_coding
minimap: chr15_KI270727v1_random:296,519-448,247
Cactus only partially aligns the alt/random
chr15 and random have GRC incidents
** chr15:19560997-19573949	OR4M2	0	+	OR4M2	HGNC:15373	ENSG00000274102.2	protein_coding
** chr16:32640154-32643358	TP53TG3D	0	+	TP53TG3D	HGNC:44657	ENSG00000205456.11	protein_coding
** chr16:33087988-33091193	TP53TG3C	0	-	TP53TG3C	HGNC:42962	ENSG00000205457.11	protein_coding
** chr16:33607936-33611141	TP53TG3	0	+	TP53TG3	HGNC:30759	ENSG00000183632.14	protein_coding
** chr16:33607936-33611141	TP53TG3B	0	+	TP53TG3B	HGNC:37202	ENSG00000261509.6	protein_coding
** chr16:33608541-33611141	TP53TG3F	0	+	TP53TG3F	HGNC:51817	ENSG00000278848.2	protein_coding
** chr16:33608541-33610516	TP53TG3E	0	+	TP53TG3E	HGNC:51816	ENSG00000275034.2	protein_coding
** chr20:33189587-33200971	DEFB124	0	-	DEFB124	HGNC:18104	ENSG00000180383.3	protein_coding
** chr21:12689317-12721070	POTED	0	+	POTED	HGNC:23822	ENSG00000166351.10	protein_coding
GRCh38 error with fixe
** chr22:19310164-19318409	DGCR6	0	+	DGCR6	HGNC:2846	ENSG00000183628.14	protein_coding
** chr22:19310374-19351909	AC007326.4	0	+	AC007326.4	NULL	ENSG00000283809.1	protein_coding
** chr22:19316947-19340726	PRODH	0	-	PRODH	HGNC:9453	ENSG00000100033.16	protein_coding
** chr4:52811-87731	ZNF595	0	+	ZNF595	HGNC:27196	ENSG00000272602.6	protein_coding
** chr4:162237470-162383089	C4orf45	0	-	C4orf45	HGNC:26342	ENSG00000164123.7	protein_coding
doesn't hit unplaced
** chr9:40658019-40723273	CBWD5	0	-	CBWD5	HGNC:24584	ENSG00000147996.16	protein_coding

* future:
** chr1:129,035,884-129,059,013 [bogus]
- repetitive array of isoseq 
- 1q21.1 - pass on to David and Colleen
- MD:the PB signal is less clear
  augustus is extending theses in some kind of interesting ways
  It is pretty easy for a gene finding to make up an ORF in a GC region
** chr13:9,841,861-9,855,951
- no cat but cactus
** better alignment mapping with kmers
*** 2020-09-13
- mydennis Arang has generated unique markers for the 0727 work: team-curation/marker_assisted/ and she has suggested running this pipeline to filter alignments: https://github.com/arangrhie/T2T-Polish/tree/master/marker_assisted
- Karen Miga Yes, this is what we are using for the pro-seq data and CHIP seq analysis (21-mers)
  Arang shared a short bash script with me for marker assisted mapping
  I have not run it yet, but Savannah has/is using it for proseq
- mydennis For us, we need to work out the Hi-C mapping that Adam ran - let's take a peek and see how it's working within segdup and duplicated regions
- karen https://github.com/arangrhie/T2T-Polish/blob/master/marker_assisted/single_copy_filter.sh

* interesting stuff
** chr22:1,093-476,524
- chr22:270,477-358,961: weird region not in repeat masker that is unique DNA 
- http://genome.ucsc.edu/s/Markd/chr22_centro_weird
KM: Can we run some kind of “shape” code like what our friend Katie Pollard was doing?
https://www.cell.com/cell-systems/pdfExtended/S2405-4712(18)30475-7



** chr11:1,971,292-2,184,069
- containing MRPL3
- region alignments to both unplaced region and primary in GRCh38,
  shown in cactus and lastz (not minimap)
- 5' adjacent TNNT3 only in hg38 primary
- theory, grch38 unplaced is wrong
- grch38 has segdup
- Known problem in GRCh38!!!

** on unplaced in GRCh38
*** Justin Zook hi @Ann Mc Cartney and @Mark Diekhans -  I've found a few more genes that seem to be duplicated in CHM13 relative to GRCh38, potentially resolving some issues. Here are Valerie's responses to what she/GRC know about the genes:
Chr17: MAP2K3 region. The relevant GRC ticket here is HG-2284, but there's been work on this region since GRCh37. The MAP2K3 gene and the nearby KCNJ12 gene are part of a segmental duplication that was largely missing in GRCh37, and for which we added partial representation in GRCh38 (adding representation for KCNJ18). However there is still an assembly gap in GRCh38 (~21.79-21.81 Mb), and we are aware that we didn't capture the full segmental duplication. As part of our most recent curation push to close gaps, I'm happy to report that Milinn, one of the WashU curators, was able to identify a collection of WGS contigs from the CHM1 GCA_001297185.2 assembly (with a little HuRef WGS and couple of fosmids) that completely spanned the gap, adding 732Kb sequence (the 50Kb gap in GRCh38 reflects default sizing). I aligned a MAP2K3 transcript (NM_145109.3) across this gap-closing path and see a partial hit on the WGS contig LJII02000447.1, suggesting there is either another MAP2K3 gene/pseudogene copy that falls in the gap. This gap-spanning fix will be released as part of GRCh38.p14, which we are aiming to release sometime shortly after ASHG.
Chr3: ZNF717 and FRG2C region. The relevant ticket here is HG-1260. We also had a report on GRCh37 about a possible collapse or alternate haplotype in this region from RefSeq. Review prior to the release of GRCh38 showed that there's a component join near here between AC108724.4 and AC133123.3 that is comprised entirely of repeat, and we were concerned that we might have dropped out some repeat copies. This is also known to be a regions with seg dups. However, the data we have suggests that the GRCh38 join is good: we've got CHM1 clones that span the join and the alignment looks good in the region of the overlap (picture below). However, the pattern of mismatches in the alignment is unusual- there are some very high quality regions and some low quality regions. B/c those pattern switches occur within the middle of GRCh38 components (and not at component boundaries) I don't think that the GRCh38 path here is wrong. Given your observations, and those of RefSeq, I think it's more likely that this is a case where there's variability in the population for this region. I've flagged the ticket for another round of review, now that there are more sequences available, to see if we can understand alternate representations and consider whether an alt would be warranted.
Chr7: MUC3A/MUC12. The relevant ticket here is HG-2313 (although the problems for this region have been known for a long time- all the way back to build 36, with ticket HG-15!). For GRCh38, we improved the representation for MUC3A, but we also received a report from RefSeq after the release of GRCh38 that the central exon of MUC12 (which has a nasty tandem repeat) was collapsed. Thanks to your alerting us, I took a look at sequence that's now available, and found we now have a CHM1 clone (AC256220.1) that appears to correct the issue.  I've pinged the ticket to make sure this fix makes it into GRCh38.p14.
**** Ann Mc Cartney
Wow Justin, this is really neat! Was was your methods of identifying these problematic genes? :slightly_smiling_face:
**** Justin Zook
We simulated perfect short reads from chm13, aligned them to GRCh38, called variants, and looked at genes in GRCh38 that had the largest number of heterozygous variants, indicating a potential duplication in chm13 relative to GRCh38.  I'm still trying to figure out a way to do this more comprehensively, but it's been good for
